src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,122 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_SIZE = 10;
+    private int currentSize = INITIAL_SIZE;
+    private int size = 0;
+    private T[] dataArray = (T[]) new Object[currentSize];
+
+    private T[] grow() {
+        double newLength = dataArray.length * 1.5;
+        currentSize = (int) newLength;
+        T[] newArray = (T[]) new Object[currentSize];
+        for (int i = 0; i < dataArray.length; i++) {
+            newArray[i] = dataArray[i];
+        }
+        return newArray;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= currentSize) {
+            dataArray = grow();
+        }
+        dataArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size >= currentSize) {
+            dataArray = grow();
+        }
 
+        if (index > size || index >= dataArray.length || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index większy niż romizar tablicy");
+        } else if (index <= size) {
+            size++;
+            for (int i = dataArray.length - 1; i > index; i--) {
+                dataArray[i] = dataArray[i - 1];
+            }
+        }
+        dataArray[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
+        int newLenght = size + list.size();
+        T[] result = (T[]) new Object[newLenght];
+
+        for (int i = 0; i < size; i++) {
+            result[i] = dataArray[i];
+        }
 
+        for (int i = size; i < newLenght; i++) {
+            result[i] = list.get(i - size);
+        }
+        dataArray = result;
+        size = newLenght;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index > dataArray.length || index < 0) {
```

Comment:
please create another method for this check

Reply:
ok, done